src/main/java/fi/fmi/avi/model/swx/amd79/immutable/SpaceWeatherRegionImpl.java

```diff
@@ -67,6 +79,118 @@ public static Builder fromAmd82(final fi.fmi.avi.model.swx.amd82.SpaceWeatherReg
                             SpaceWeatherLocation.valueOf(locationIndicator.name())));
         }
 
+        private static Geometry buildGeometry(final double minLatitude, final double minLongitude, final double maxLatitude, final double maxLongitude) {
+            final double absMinLongitude = Math.abs(minLongitude);
+            final double absMaxLongitude = Math.abs(maxLongitude);
+            return PolygonGeometryImpl.builder()
+                    .setCrs(CoordinateReferenceSystemImpl.wgs84())
+                    .mutateExteriorRingPositions(coordinates -> {
+                        if (absMinLongitude == 180d && absMaxLongitude == 180d) {
+                            addExteriorRingPositions(coordinates, minLatitude, -180d, maxLatitude, 180d);
+                        } else if (absMinLongitude == 180d) {
+                            addExteriorRingPositions(coordinates, minLatitude, -180d, maxLatitude, maxLongitude);
+                        } else if (absMaxLongitude == 180d) {
+                            addExteriorRingPositions(coordinates, minLatitude, minLongitude, maxLatitude, 180d);
+                        } else {
+                            addExteriorRingPositions(coordinates, minLatitude, minLongitude, maxLatitude, maxLongitude);
+                        }
+                    })
+                    .build();
```

Comment:
I know this was only moved from the TAC parser, but I found that this could be simplified to:

```suggestion
            final double normalizedMinLongitude = Math.abs(minLongitude) == 180d ? -180d : minLongitude;
            final double normalizedMaxLongitude = Math.abs(maxLongitude) == 180d ? 180d : maxLongitude;
            return PolygonGeometryImpl.builder()
                    .setCrs(CoordinateReferenceSystemImpl.wgs84())
                    .mutateExteriorRingPositions(coordinates ->
                            addExteriorRingPositions(coordinates, minLatitude, normalizedMinLongitude, maxLatitude, normalizedMaxLongitude))
                    .build();
```

Though some unit tests would be nice before applying this change, I may have been mistaken.